tmt/base.py

```diff
@@ -2908,6 +2908,28 @@ def to_serialized(self) -> Dict[str, Any]:
     def from_serialized(cls, serialized: Dict[str, Any]) -> 'Result':
         return cls(serialized, name=serialized.pop('name'))
 
+    @staticmethod
+    def failures(log: Optional[str], msg_type: str = 'FAIL') -> str:
+        """ Filter stdout and get only messages with certain type """
+        if not log:
+            return ''
+        filtered = ''
+
+        # Filter beakerlib style logs
+        for m in re.findall(
+                fr'(^.*\[\s*{msg_type}\s*\][\S\s]*?)(?:^::\s+\[[0-9: ]+|:{{80}})',
+                '\n'.join(log.split('\n')[::-1]), re.MULTILINE):
+            filtered += m.strip() + '\n'
+        if filtered:
+            return '\n'.join(filtered.strip().split('\n')[::-1])
+
+        # Check for other failures and errors when not using beakerlib
+        for m in re.findall(
+                f'.*(?=error|fail|{msg_type}).*', log, re.IGNORECASE | re.MULTILINE):
```

Comment:
Maybe we should only look for whole words? Currently this matches strings as `pipefail` as well.